tests/src/Kernel/ContentLanguageSwitcherTest.php

```diff
@@ -0,0 +1,215 @@
+
+namespace Drupal\Tests\oe_whitelabel\Kernel;
+
+use Drupal\node\Entity\Node;
+use Drupal\KernelTests\KernelTestBase;
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Test content language switcher rendering.
+ */
+class ContentLanguageSwitcherTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'block',
+    'content_translation',
+    'language',
+    'locale',
+    'node',
+    'oe_bootstrap_theme_helper',
+    'oe_corporate_blocks',
+    'oe_multilingual',
+    'oe_whitelabel_helper',
+    'oe_whitelabel_multilingual',
+    'system',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('user');
+    $this->installSchema('system', 'sequences');
+    $this->installSchema('user', ['users_data']);
+    $this->installConfig(['user']);
+
+    $this->installSchema('locale', [
+      'locales_location',
+      'locales_source',
+      'locales_target',
+    ]);
+
+    $this->installEntitySchema('node');
+    $this->installSchema('node', 'node_access');
+    $this->installConfig([
+      'content_translation',
+      'language',
+      'locale',
+      'oe_multilingual',
+      'oe_whitelabel_helper',
+      'system',
+    ]);
+    $this->container->get('module_handler')->loadInclude('oe_multilingual', 'install');
+    oe_multilingual_install(FALSE);
+
+    $this->container->get('theme_installer')->install(['oe_whitelabel']);
+    $this->config('system.theme')->set('default', 'oe_whitelabel')->save();
+
+    // Call the installation hook of the User module which creates the
+    // Anonymous user and User 1. This is needed because the Anonymous user
+    // is loaded to provide the current User context which is needed
+    // in places like route enhancers.
+    // @see CurrentUserContext::getRuntimeContexts().
+    // @see EntityConverter::convert().
+    module_load_include('install', 'user');
+    user_install();
+
+    \Drupal::service('kernel')->rebuildContainer();
+  }
+
+  /**
+   * Tests the rendering of the language switcher block.
+   */
+  public function testMultilingualLanguageSwitcherBlockRendering(): void {
+    $node = Node::create([
+      'title' => 'Hello, world!',
+      'type' => 'oe_demo_translatable_page',
+    ]);
+    /** @var \Drupal\Core\Entity\EntityInterface $translation */
+    $node->addTranslation('es', ['title' => '¡Hola mundo!'])->save();
+
+    // Simulate a request to the canonical route of the node with Bulgarian
+    // language prefix.
+    $this->setCurrentRequest('/bg/node/' . $node->id());
+
+    // Setup and render language switcher block.
+    $block_manager = \Drupal::service('plugin.manager.block');
+    $config = [
+      'id' => 'oe_multilingual_content_language_switcher',
+      'label' => 'Content language switcher',
+      'provider' => 'oe_multilingual',
+      'label_display' => '0',
+    ];
```

Comment:
We can remove all array keys except 'label_display'. They will be filled with default values.

Reply:
Ok, perhaps we want all the keys to fully simulate a block with the typical configuration - ok for me.